ledger/internal/eval_blackbox_test.go

```diff
@@ -989,133 +990,186 @@ func TestLogsInBlock(t *testing.T) {
-	ephemeral := []txntest.Txn{
-		{
+		benefactor := txntest.Txn{
 			Type:     "pay",
-			Amount:   0,
-			Sender:   ghost,
-			Receiver: ghost,
-			Fee:      0,
-		},
-		{
-			Type:          "axfer",
-			AssetAmount:   0,
-			Sender:        ghost,
-			AssetReceiver: basics.Address{0x02},
-			XferAsset:     basics.AssetIndex(1),
-			Fee:           0,
-		},
-		{
-			Type:          "axfer",
-			AssetAmount:   0,
-			Sender:        basics.Address{0x0c, 0x0b, 0x0a, 0x0c},
-			AssetReceiver: addrs[0],
-			AssetSender:   addrs[1],
-			XferAsset:     asaIndex,
-			Fee:           0,
-		},
-		{
-			Type:          "afrz",
-			Sender:        basics.Address{0x0f, 0x0e, 0xe, 0xe},
-			FreezeAccount: addrs[0], // creator, therefore is opted in
-			FreezeAsset:   asaIndex,
-			AssetFrozen:   true,
-			Fee:           0,
-		},
-		{
-			Type:          "afrz",
-			Sender:        basics.Address{0x0f, 0x0e, 0xe, 0xe},
-			FreezeAccount: addrs[0], // creator, therefore is opted in
-			FreezeAsset:   asaIndex,
-			AssetFrozen:   false,
-			Fee:           0,
-		},
-	}
+			Sender:   addrs[0],
+			Receiver: addrs[0],
+			Fee:      2000,
+		}
 
-	for i, e := range ephemeral {
-		eval = nextBlock(t, l)
-		err := txgroup(t, l, eval, &benefactor, &e)
-		require.NoError(t, err, "i=%d %s", i, e.Type)
-		endBlock(t, l, eval)
-	}
+		ephemeral := []txntest.Txn{
+			{
+				Type:     "pay",
+				Amount:   0,
+				Sender:   ghost,
+				Receiver: ghost,
+				Fee:      0,
+			},
+			{ // Axfer of 0
+				Type:          "axfer",
+				AssetAmount:   0,
+				Sender:        ghost,
+				AssetReceiver: basics.Address{0x02},
+				XferAsset:     basics.AssetIndex(1),
+				Fee:           0,
+			},
+			{ // Clawback
+				Type:          "axfer",
+				AssetAmount:   0,
+				Sender:        ghost,
+				AssetReceiver: addrs[0],
+				AssetSender:   addrs[1],
+				XferAsset:     asaIndex,
+				Fee:           0,
+			},
+			{ // Freeze
+				Type:          "afrz",
+				Sender:        ghost,
+				FreezeAccount: addrs[0], // creator, therefore is opted in
+				FreezeAsset:   asaIndex,
+				AssetFrozen:   true,
+				Fee:           0,
+			},
+			{ // Unfreeze
+				Type:          "afrz",
+				Sender:        ghost,
+				FreezeAccount: addrs[0], // creator, therefore is opted in
+				FreezeAsset:   asaIndex,
+				AssetFrozen:   false,
+				Fee:           0,
+			},
+			{ // App call
+				Type:          "appl",
+				Sender:        ghost,
+				ApplicationID: basics.AppIndex(2),
+				Fee:           0,
+			},
+			{ // App creation (only works because it's also deleted)
+				Type:         "appl",
+				Sender:       ghost,
+				OnCompletion: transactions.DeleteApplicationOC,
+				Fee:          0,
+			},
+		}
+
+		// v33 is the likely version for UnfundedSenders. Change if that doesn't happen.
+		if ver < 33 {
+			// In the old days, balances.Move would try to increase the rewardsState on the unfunded account
+			for i, e := range ephemeral {
+				dl.txgroup("balance 0 below min", benefactor.Noted(strconv.Itoa(i)), &e)
```

Comment:
maybe use the same function as below to ensure both branches check the same code?